paddle/fluid/operators/conv_cudnn_helper.h

```diff
@@ -568,69 +566,141 @@ struct SearchAlgorithm<cudnnConvolutionBwdFilterAlgoPerf_t> {
-                            workspace_size_limit));
-              };
-              workspace_handle.RunFuncSync(cudnn_find_func,
-                                           workspace_size_limit);
-
-              VLOG(3)
-                  << "BwdFilterAlgo Perf result: (algo: stat, time, memory)";
-              for (int i = 0; i < returned_algo_count; ++i) {
-                const auto& stat = perf_stat[i];
-                VLOG(3) << stat.algo << ": " << stat.status << " " << stat.time
-                        << " " << stat.memory;
-              }
-              return perf_stat[0].algo;
-            });
-      } else {
-        auto max_algos = MaxBwdFilterAlgos(args.handle);
-        algo = algo_cache.GetAlgorithm(
-            x_dims, w_dims, args.s, args.p, args.d, 0,
-            static_cast<int64_t>(args.cudnn_dtype), [&]() {
-              algo_t chosen_algo;
-              std::vector<perf_t> perf_results(max_algos);
-              int actual_algos = 0;
-              PADDLE_ENFORCE_CUDA_SUCCESS(
-                  platform::dynload::
-                      cudnnFindConvolutionBackwardFilterAlgorithm(
-                          args.handle, args.idesc.desc(), args.odesc.desc(),
-                          args.cdesc.desc(), args.wdesc.desc(),
-                          perf_results.size(), &actual_algos,
-                          perf_results.data()));
-              perf_results.resize(actual_algos);
-              ChooseAlgo<perf_t, algo_t>(perf_results, workspace_size_limit,
+                            perf_results.size(), &actual_algos,
+                            perf_results.data()));
+                perf_results.resize(actual_algos);
+                ChooseAlgo<PerfT, AlgoT>(perf_results, workspace_size_limit,
                                          &chosen_algo);
-              return chosen_algo;
-            });
+                return framework::CudnnConvAlgorithm<AlgoT>(
+                    chosen_algo, GetWorkspaceSize(args, chosen_algo));
+              });
+          algo = ret.algo;
+        }
       }
     }
     VLOG(3) << "choose algo " << algo;
     return algo;
   }
 
-  static size_t GetWorkspaceSize(const ConvArgs& args, algo_t algo) {
-    size_t workspace_size = 0;
-    PADDLE_ENFORCE_CUDA_SUCCESS(
-        platform::dynload::cudnnGetConvolutionBackwardFilterWorkspaceSize(
-            args.handle, args.idesc.desc(), args.odesc.desc(),
-            args.cdesc.desc(), args.wdesc.desc(), algo, &workspace_size));
-    return workspace_size;
+  static size_t GetWorkspaceSize(const ConvArgs& args, AlgoT algo) {
+    auto& algo_cache =
+        *(framework::ConvSearchCache::Instance().GetBackwardFilter());
+
+    auto x_dims = framework::vectorize(args.x->dims());
+    auto w_dims = framework::vectorize(args.w->dims());
+
+    auto ret = algo_cache.Find(x_dims, w_dims, args.s, args.p, args.d, 0,
+                               static_cast<int64_t>(args.cudnn_dtype));
```

Comment:
Does `GetWorkspaceSize` inside `FindAlgorithmHeuristic` need to find in cache?

Reply:
Yes. If I want to avoid finding in cache, I need add a new argument.